os/arch/arm/src/armv7-r/arm_initialstate.c

```diff
@@ -100,7 +100,9 @@
 
 FAR int up_setup_guard_region(FAR struct tcb_s *tcb, uint8_t ttype)
 {
+#ifdef CONFIG_MPU_STACKGUARD
```

Comment:
I guess, whole function should go under MPU_STACKGUARD flag as no one else uses this API and also There is no use of ttype inside this API, So we can pass only one argument and change the function definition accordingly.